eZ/Publish/Core/Repository/UserService.php

```diff
@@ -617,7 +617,13 @@ public function deleteUser(APIUser $user): iterable
 
         $this->repository->beginTransaction();
         try {
-            $affectedLocationIds = $this->repository->getContentService()->deleteContent($loadedUser->getVersionInfo()->getContentInfo());
+            foreach ($this->userHandler->loadRoleAssignmentsByGroupId($user->id) as $roleAssignment) {
+                $this->userHandler->removeRoleAssignment($roleAssignment->id);
+            }
+
```

Comment:
@alongosz It works properly for users now, but we should do the same thing for User Groups. 

Deleting users properly invalidates cache now.

Reply:
> @alongosz It works properly for users now, but we should do the same thing for User Groups.
> 
> Deleting users properly invalidates cache now.

Added via [713a3ad](https://github.com/ezsystems/ezplatform-kernel/pull/374/commits/713a3ad703eaf068a6707ecef5fabbbf8dc8c248).

Reply:
@alongosz hmm, it looks like it's not working. A count is not invalidated after deleting a user group.

It looks like the `deleteUserGroup` method is not used in the Back Office, but only in REST API. In the Back Office, User Group is treated as a 'normal' content, so it's basically trashed, hence, the `deleteUserGroup` method is not used.

Reply:
> @alongosz hmm, it looks like it's not working. A count is not invalidated after deleting a user group.
> 
> It looks like the `deleteUserGroup` method is not used in the Back Office, but only in REST API. In the Back Office, User Group is treated as a 'normal' content, so it's basically trashed, hence, the `deleteUserGroup` method is not fired.

What's going to happen if you remove it from trash? Usually AdminUI should use dedicated strategy for user-related items.

Reply:
Nothing really happens. It has a dedicated strategy only for Users, not for User Groups.

Reply:
> Nothing really happens. It has a dedicated strategy only for Users, not for User Groups.

Hmm, perhaps it requires fixing on AdminUI side. User Group should be deleted using User API, not Content API. So basically, for the fix you're working on, does this mean that if user group is assigned and gets removed via AdminUI, the count is incorrect and it affects the fix?

Reply:
Yes, this is exactly the case. Obviously, it breaks only if we are using caching, so it works properly for the original solution without caching count and paginated list. 

Changing the way how we are deleting User Groups shouldn't be a bit of a design change? We would basically switch design here and this could be potentially breaking for some customers used to have groups in trash for instance.